Source/WebGPU/WebGPU/HardwareCapabilities.mm

```diff
@@ -419,8 +420,28 @@ static WGPULimits mergeLimits(const WGPULimits& previous, const WGPULimits& next
     };
 }
 
+static bool isPhysicalHardware()
+{
+#if PLATFORM(IOS_FAMILY_SIMULATOR)
+    return false;
+#else
+    static bool result = true;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        uint32_t isVM = 0;
+        size_t size = sizeof(isVM);
+        if (!sysctlbyname("kern.hv_vmm_present", &isVM, &size, NULL, 0))
+            result = isVM ? [[NSUserDefaults standardUserDefaults] boolForKey:@"WebKitAllowWebGPUOnVMs"] : true;
```

Comment:
Have you verified that this works? User defaults are quite restricted in WebKit helper processes, and we quite prefer piping settings from the UI process.

Reply:
Yes I confirmed it works via `defaults write -g WebKitAllowWebGPUOnVMs 1`